### PR TITLE
docs: add genfile initial docs/spec

### PR DESCRIPTION
**Pull request by katcipis**
Initial spec for the new generate_file block.

### CODE REVIEW

File: docs/codegen/generate-file.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+* Terramate Metadata references
+* Terramate function calls
+* Expressions using string interpolation
+
+The final evaluated **content** attribute **must** be a string.
+
+Multiple `generate_file` blocks with the same label/filename will
+result in an error.
+
+Here is an example on how a JSON file can be created:
```
Comment (soerenmartius) [inline docs/codegen/generate-file.md:25]:
```suggestion
# File Generation

Terramate supports the generation of arbitrary text files referencing 
[Terramate defined data](../sharing-data.md).

File generation is done using `generate_file`
blocks in [Terramate configuration files](../config-overview.md).

The block **must** have a single label, that will be used to determine the
name of the generated file. Inside the block, the **content** attributes define
the string that will be written on the file.

The definition of the **content** attribute may include:

* Terramate Global references
* Terramate Metadata references
* Terramate function calls
* Expressions using string interpolation

The final evaluated **content** attribute **must** be a string.

Multiple `generate_file` blocks with the same label/filename will
result in an error.

Here is an example of how a JSON file can be created:
```

File: docs/codegen/overview.md [hunk 2]
```diff
@@ -4,5 +4,7 @@ Code generation is the main way you can glue [Terramate data](../sharing-data.md
 to other tools, like Terraform.
 
 Different code generation strategies will be provided in the future to support
-each integration scenario the best way possible. Currently we support
-[HCL generation](./generate-hcl.md).
+each integration scenario the best way possible. Currently we support:
```
Comment (soerenmartius) [inline docs/codegen/overview.md:7]:
```suggestion
each integration scenario in the best way possible. Currently, we support:
```

**Review by soerenmartius — CHANGES_REQUESTED** [review]

**Review by soerenmartius — APPROVED** [review]
lgtm! Good job!

**Review by i4ki — APPROVED** [review]
LGTM :rocket:

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**